pkg/test/framework/components/policybackend/kube.go

```diff
@@ -74,6 +74,8 @@ spec:
       - name: app
         image: "{{.Hub}}/test_policybackend:{{.Tag}}"
         imagePullPolicy: {{.ImagePullPolicy}}
+        securityContext:
+          runAsUser: 1
```

Comment:
The mechanism behind PSP are a little bit different since if you use a Deployment the `ServiceAccount` of the deployment needs the rights to use the privileged PSP. In our setup all `ServiceAccounts` in `istio-system` have the appropriate rights to use the privileged PSP. During the test some namespaces will be created all `ServiceAccounts` in these namespaces only have the rights to use the `restricted` PSP. Since the the namespaces are generated dynamically we can't preprovision the according Rules. On the other hand I don't see any value in running the example test backend as root (it also runs perfectly as `daemon` user).

Reply:
Ah makes sense, thanks! Yeah I don't think they need root like you said